Client/Frontend/Browser/Playlist/Cells/PlaylistCell.swift

```diff
@@ -0,0 +1,169 @@
+        contentView.addSubview(iconStackView)
+        contentView.addSubview(infoStackView)
+        iconStackView.addArrangedSubview(thumbnailHolder)
+        infoStackView.addArrangedSubview(titleLabel)
+        infoStackView.addArrangedSubview(detailLabel)
+        contentView.addSubview(separator)
+        thumbnailHolder.addSubview(thumbnailView)
+        
+        thumbnailHolder.snp.makeConstraints {
+            // Keeps a 94.0px width on iPhone-X as per design
+            $0.width.equalTo(iconStackView.snp.height).multipliedBy(1.46875 /* 94.0 / (tableViewCellHeight - (8.0 * 2)) */)
+            $0.height.equalToSuperview()
+        }
+        
+        thumbnailView.snp.makeConstraints {
+            $0.center.equalToSuperview()
+            $0.left.right.top.bottom.equalToSuperview().priority(.high)
+            $0.width.height.equalToSuperview()
+        }
+        
+        iconStackView.snp.makeConstraints {
+            $0.leading.equalToSuperview().offset(12.0)
+            $0.top.bottom.equalToSuperview().inset(8.0)
+        }
+        
+        infoStackView.snp.makeConstraints {
+            $0.left.equalTo(iconStackView.snp.right).offset(8.0)
+            $0.right.equalToSuperview().offset(-15.0)
+            $0.centerY.equalToSuperview()
+            $0.top.greaterThanOrEqualTo(iconStackView.snp.top)
+            $0.bottom.lessThanOrEqualTo(iconStackView.snp.bottom)
+        }
+        
+        separator.snp.makeConstraints {
+            $0.left.equalTo(titleLabel.snp.left)
+            $0.right.bottom.equalToSuperview()
+            $0.height.equalTo(1.0 / UIScreen.main.scale)
+        }
+    }
+    
+    private static func onThumbnailChanged(_ imageView: PlaylistResizingThumbnailView) {
+        guard let superView = imageView.superview else { return }
+        
+        imageView.snp.remakeConstraints {
+            $0.center.equalToSuperview()
+            
+            if let size = imageView.image?.size {
+                if size.width > superView.bounds.width || size.height > superView.bounds.height {
+                    $0.width.height.equalToSuperview()
+                } else {
+                    $0.left.right.top.bottom.equalToSuperview().priority(.high)
+                    $0.width.height.equalTo(28.0)
+                }
+            } else {
+                $0.width.height.equalToSuperview()
+            }
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override var layoutMargins: UIEdgeInsets {
+        get {
+            return .zero
+        }
+
+        set { //swiftlint:disable:this unused_setter_value
+            super.layoutMargins = .zero
+        }
+    }
+    
+    override var separatorInset: UIEdgeInsets {
+        get {
+            return UIEdgeInsets(top: 0, left: self.titleLabel.frame.origin.x, bottom: 0, right: 0)
+        }
+        
+        set (newValue) {
+            _ = newValue
```

Comment:
minor nit: disable swiftlint like above (`// swiftlint:disable:this unused_setter_value`)